packages/smooth_app/lib/pages/product/add_new_product_page.dart

```diff
@@ -163,59 +207,53 @@ class _AddNewProductPageState extends State<AddNewProductPage>
-          ),
-        );
-        if (leaveThePage == true) {
-          AnalyticsHelper.trackEvent(
-            widget.events[EditProductAction.leaveEmpty]!,
-            barcode: barcode,
-          );
-        }
-        return leaveThePage ?? false;
-      },
+      onWillPop: _onWillPop,
       child: SmoothScaffold(
-        appBar: SmoothAppBar(
-          title: ListTile(
-            title: Text(
-              upToDateProduct.productName ?? appLocalizations.new_product,
-            ),
-            subtitle: Text(barcode),
-          ),
-        ),
-        body: Padding(
-          padding: const EdgeInsetsDirectional.symmetric(
-            vertical: VERY_LARGE_SPACE,
-          ),
-          child: SingleChildScrollView(
-            child: Column(
-              crossAxisAlignment: CrossAxisAlignment.start,
-              children: <Widget>[
-                if (widget.displayPictures) _buildCard(_getImageRows(context)),
-                _buildCard(_getNutriscoreRows(context)),
-                _buildCard(_getEcoscoreRows(context)),
-                _buildCard(_getNovaRows(context)),
-                if (widget.displayMisc) _buildCard(_getMiscRows(context)),
-                const SizedBox(height: MINIMUM_TOUCH_SIZE),
-              ],
-            ),
+        body: SafeArea(
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: <Widget>[
+              Container(
+                decoration: BoxDecoration(
+                  gradient: LinearGradient(
+                    colors: <Color>[
+                      Theme.of(context).colorScheme.inversePrimary,
+                      Theme.of(context).colorScheme.secondary
+                    ],
+                  ),
+                ),
+                child: FAProgressBar(
+                  animatedDuration: SmoothAnimationsDuration.short,
+                  backgroundColor: Theme.of(context).colorScheme.secondary,
+                  size: 8,
+                  currentValue: _progress,
+                  maxValue: 1,
+                  progressColor: Theme.of(context).colorScheme.inversePrimary,
+                ),
+              ),
+              _backButton(),
+              Expanded(
+                child: PageView(
+                  controller: _pageController,
+                  children: <Widget>[
+                    if (widget.displayPictures)
+                      _buildCard(_getImageRows(context)),
+                    _buildCard(_getNutriscoreRows(context)),
+                    _buildCard(_getEcoscoreRows(context)),
+                    _buildCard(_getNovaRows(context)),
+                    if (widget.displayMisc) _buildCard(_getMiscRows(context)),
+                  ],
+                ),
+              ),
+              Card(
+                margin: EdgeInsets.zero,
+                elevation: 15.0,
+                child: SizedBox(
+                  height: MediaQuery.of(context).size.height * 0.1,
```

Comment:
Now that we use the latest version of Flutter, you can simplify by using `MediaQuery.sizeOf()`